### PR TITLE
MAUI: Update authentication samples

### PR DESCRIPTION
**Pull request by ZackAllen**
# Description

- Fixed OAuth workflow on all MAUI platforms.
- Updated UI and functionality for `Create and save map` and ` Search for webmap`. 
- Fixed functionality and updated UI for `Token secured challenge`.

## Type of change

- [x] Bug fix

**Platforms tested on**:

- [x] WinUI
- [x] Android
- [x] iOS
- [x] MacCatalyst

## Checklist

- [x] Runs and compiles on all active platforms
- [ ] Legacy platforms still compile and run (if applicable)
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [ ] `sample_sync.py` runs without making changes
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] Code is commented with correct formatting
- [x] All variable and method names are good and make sense
- [x] There is no leftover commented code
- [ ] Screenshots are correct size and display in description tab

### CODE REVIEW

**Review by duffh — APPROVED** [review]

File: src/MAUI/Maui.Samples/Helpers/ArcGISLoginPrompt.cs [hunk 1]
```diff
@@ -124,51 +122,54 @@ public static async Task<Credential> PromptCredentialAsync(CredentialRequestInfo
...
 #endif
+    }
 
 #if ANDROID
 
     [Activity(NoHistory = true, Exported = true, LaunchMode = LaunchMode.SingleTop)]
     [IntentFilter(new[] { Intent.ActionView },
        Categories = new[] { Intent.CategoryDefault, Intent.CategoryBrowsable },
-       DataScheme = "maui-samples-app", DataHost = "auth")]
+       DataScheme = "maui-ags-app", DataHost = "auth")]
```
Comment (amweis226) [inline src/MAUI/Maui.Samples/Helpers/ArcGISLoginPrompt.cs:172]: Why the change from "maui-samples-app" to "maui-ags-app"?

File: src/MAUI/Maui.Samples/Samples/Map/AuthorMap/AuthorMap.xaml.cs [hunk 2]
```diff
@@ -33,13 +27,13 @@ public partial class AuthorMap : ContentPage
         private const string ArcGISOnlineUrl = "https://www.arcgis.com/sharing/rest";
 
         // String array to store names of the available basemaps
```
Comment (amweis226) [inline src/MAUI/Maui.Samples/Samples/Map/AuthorMap/AuthorMap.xaml.cs:29]: Many comments had periods added to the end of them, but this comment and the one on line 39 did not. Is this intentional (perhaps to differentiate between preliminary declarations and instructions)? This seems inconsistent across samples. For example, Browse Building Floors has periods at the end of every comment, but Display Map has none. Might be worth considering a separate PR for establishing consistency across comments, would also make such changes more relevant.

**Comment by amweis226** [comment]
The apps and code look good just a couple nitpicks:

ArcGIS token challenge
    - In ArcGIS Pro, when private content is authorized for access and added to a map, the MapView will typically zoom to the extent of the layer. Might be worth considering as user friendliness and similarity to Pro spec could reasonably justify adding this.

Authenticate with OAuth
    - In _About_ under _How to use the sample_, "premium content" implies that there are types of contents associated with pay tiers. Maybe "private content" would be more accurate?
